notebooks/exercise7-oqmd-optimade-tutorial.ipynb

```diff
@@ -0,0 +1,645 @@
+      "Success!\n",
+      "\n",
+      "Querying: http://oqmd.org/optimade/structures?filter=stability%3C%3D0+AND+elements+HAS+%22O%22+AND+nelements%3D3+AND+_oqmd_band_gap%3E0&page_limit=200&page_offset=200&response_fields=id%2C_oqmd_entry_id%2Clattice_vectors%2Ccartesian_site_positions%2Cspecies_at_sites%2C_oqmd_band_gap\n",
+      "Success!\n",
+      "\n",
+      "Querying: http://oqmd.org/optimade/structures?filter=stability%3C%3D0+AND+elements+HAS+%22O%22+AND+nelements%3D3+AND+_oqmd_band_gap%3E0&page_limit=200&page_offset=400&response_fields=id%2C_oqmd_entry_id%2Clattice_vectors%2Ccartesian_site_positions%2Cspecies_at_sites%2C_oqmd_band_gap\n",
+      "Success!\n",
+      "\n",
+      "Querying: http://oqmd.org/optimade/structures?filter=stability%3C%3D0+AND+elements+HAS+%22O%22+AND+nelements%3D3+AND+_oqmd_band_gap%3E0&page_limit=200&page_offset=600&response_fields=id%2C_oqmd_entry_id%2Clattice_vectors%2Ccartesian_site_positions%2Cspecies_at_sites%2C_oqmd_band_gap\n",
+      "Success!\n",
+      "\n",
+      "Querying: http://oqmd.org/optimade/structures?filter=stability%3C%3D0+AND+elements+HAS+%22O%22+AND+nelements%3D3+AND+_oqmd_band_gap%3E0&page_limit=200&page_offset=800&response_fields=id%2C_oqmd_entry_id%2Clattice_vectors%2Ccartesian_site_positions%2Cspecies_at_sites%2C_oqmd_band_gap\n",
+      "Success!\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We need more than 200 datapoints for machine learning - if more data is available\n",
+    "\n",
+    "# As the first step, here's the same script from the cell above, but kept inside a function \n",
+    "\n",
+    "def query_oqmd_optimade(query):\n",
+    "    print(\"\\nQuerying: {}\".format(query))\n",
+    "    response = requests.get(query)\n",
+    "    if response.status_code == 200:\n",
+    "        print(\"Success!\")\n",
+    "        return response.json()\n",
+    "    else:\n",
+    "        print(\"Query failed. Status: {}\".format(response.status_code))\n",
+    "        print(\"Error Message: {}\".format(response.text))\n",
+    "        return \n",
+    "\n",
+    "    \n",
+    "    \n",
+    "\n",
+    "# Next, we query for 1000 materials in total using 5 sequential API queries - each paginated to\n",
+    "# retrieve 200 materials\n",
+    "\n",
+    "\n",
+    "load_data_from_saved = True  \n",
+    "# This is to avoid querying OQMD repeatedly for the same data, if .\n",
+    "# Because the data I downloaded is already available as a JSON file in this Git repo.\n",
+    "\n",
+    "# But if you'd like to try out querying OQMD, set \"load_data_from_saved\" as \"False\"\n",
+    "# Querying OQMD for this particular data would take about 5-10 minutes to complete\n",
+    "\n",
+    "\n",
+    "dataset_filename = Path(__file__).parent.joinpath(\"./oqmd_exercise_data/raw_dataset_saved.json\")\n",
```

Comment:
```suggestion
"if \"__file__\" in globals():\n",
    "    dataset_filename = Path(__file__).parent.joinpath(\"./oqmd_exercise_data/raw_dataset_saved.json\")\n",
    "else:\n",
    "    dataset_filename = Path(\"__file__\").parent.joinpath(\"./oqmd_exercise_data/raw_dataset_saved.json\")\n",
    "    \n",
```
\_\_file\_\_ does not seem to work in a jupyter notebook.
And I get the error message: 
NameError: name '\_\_file\_\_' is not defined

Replacing it with "\_\_file\_\_" does work.  To not break the original code I added an if statement to distinguish between these two situations.
See also [here](https://stackoverflow.com/questions/63583062/path-problem-nameerror-name-file-is-not-defined)